realm/realm-library/src/main/cpp/objectserver_shared.hpp

```diff
@@ -19,123 +19,106 @@
-            using ProtocolError = realm::sync::ProtocolError;
-            auto protocol_error = static_cast<ProtocolError>(error_code.value());
-
-            // Documented here: https://realm.io/docs/realm-object-server/#client-recovery-from-a-backup
-            if (protocol_error == ProtocolError::bad_server_file_ident
-                || protocol_error == ProtocolError::bad_client_file_ident
-                || protocol_error == ProtocolError::bad_server_version
-                || protocol_error == ProtocolError::diverging_histories) {
-
-                // Add a SyncFileActionMetadata marking the Realm as needing to be deleted.
-                auto recovery_path = realm::util::reserve_unique_file_name(
-                        realm::SyncManager::shared().recovery_directory_path(),
-                        realm::util::create_timestamped_template("recovered_realm"));
-                auto original_path = local_realm_path;
-
-                realm::jni_util::Log::d("A client reset is scheduled for the next app start");
-                realm::SyncManager::shared().perform_metadata_update([original_path = std::move(original_path),
-                        recovery_path = std::move(recovery_path)](const auto &manager) {
-                    realm::SyncFileActionMetadata(manager,
-                                                  realm::SyncFileActionMetadata::Action::HandleRealmForClientReset,
-                                                  original_path,
-                                                  "",
-                                                  "",
-                                                  realm::util::Optional<std::string>(
-                                                          std::move(recovery_path)));
-                });
-
-            } else {
-                auto session_ref = weak_session_ref.lock();
-                if (session_ref) {
-                        session_ref.get()->call_with_local_ref([&](JNIEnv* local_env, jobject obj) {
-                            static realm::jni_util::JavaMethod notify_error_handler(
-                                    local_env, obj, "notifySessionError", "(ILjava/lang/String;)V");
-                            local_env->CallVoidMethod(
-                                    obj, notify_error_handler, error_code.value(), local_env->NewStringUTF(message.c_str()));
-                        });
-                }
+        // path on disk of the Realm file.
+        // the sync configuration object.
+        // the session which should be bound.
+        auto bind_handler = [=](const std::string& path, const SyncConfig& syncConfig, std::shared_ptr<SyncSession> session) {
+            realm::jni_util::Log::d("Callback to Java requesting token for path");
+
+            JNIEnv *env = realm::jni_util::JniUtils::get_env(true);
+            jstring access_token_string = (jstring) env->CallStaticObjectMethod(java_syncmanager,
```

Comment:
if the `CallStaticObjectMethod` fails we will not invoke any further JNI method (not even `JStringAccessor`) since `access_token_string` is not valid & the lambda execution finishes.